frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-ref.ts

```diff
@@ -0,0 +1,84 @@
+import { K8sModel } from '../../api/common-types';
+import { GetAPIVersionForModel, GetGroupVersionKindForResource } from '../../api/k8s-types';
+import { K8sGroupVersionKind, K8sResourceKindReference } from '../../extensions/console-types';
+
+/**
+ * Provides a reference string that uniquely identifies the group, version, and kind of a k8s resource.
+ * @param K8sGroupVersionKind Pass K8sGroupVersionKind which will have group, version, and kind of a k8s resource.
+ * @param K8sGroupVersionKind.group Pass group of k8s resource or model.
+ * @param K8sGroupVersionKind.version Pass version of k8s resource or model.
+ * @param K8sGroupVersionKind.kind Pass kind of k8s resource or model.
+ * @return The reference for any k8s resource i.e `group~version~kind`.
+ * If the group will not be present then "core" will be returned as part of the group in reference.
+ * * */
+export const getReference = ({
+  group,
+  version,
+  kind,
+}: K8sGroupVersionKind): K8sResourceKindReference => [group || 'core', version, kind].join('~');
+
+/**
+ * Provides a reference string that uniquely identifies the group, version, and kind of a k8s model.
+ * @param model k8s model
+ * @return The reference for model i.e `group~version~kind`.
+ * * */
+export const getReferenceForModel = (model: K8sModel): K8sResourceKindReference =>
+  getReference({ group: model.apiGroup, version: model.apiVersion, kind: model.kind });
```

Comment:
Same deal for this one as it's just a wrapper around it.

Although, do we even have a `Model => K8sGroupVersionKind` utility?

I think we might need one of those.

Reply:
added one for it and exposed via core-api `K8sModel => K8sGroupVersionKind`